cupy/core/core.pyx

```diff
@@ -1952,36 +1956,48 @@ cpdef function.Module compile_with_cache(
     if _cuda_runtime_version < 0:
         _cuda_runtime_version = runtime.runtimeGetVersion()
 
-    if 9020 <= _cuda_runtime_version < 9030:
-        bundled_include = 'cuda-9.2'
-    elif 10000 <= _cuda_runtime_version < 10010:
-        bundled_include = 'cuda-10.0'
-    elif 10010 <= _cuda_runtime_version < 10020:
-        bundled_include = 'cuda-10.1'
-    elif 10020 <= _cuda_runtime_version < 10030:
-        bundled_include = 'cuda-10.2'
-    elif 11000 <= _cuda_runtime_version < 11010:
-        bundled_include = 'cuda-11.0'
-    elif 11010 <= _cuda_runtime_version < 11020:
-        bundled_include = 'cuda-11.1'
-    else:
-        # CUDA versions not yet supported.
-        bundled_include = None
-
-    cuda_path = cuda.get_cuda_path()
-
-    if bundled_include is None and cuda_path is None:
-        raise RuntimeError(
-            'Failed to auto-detect CUDA root directory. '
-            'Please specify `CUDA_PATH` environment variable if you '
-            'are using CUDA versions not yet supported by CuPy.')
+    global _cuda_path
+    if _cuda_path == '':
+        if not _is_hip:
+            _cuda_path = cuda.get_cuda_path()
+        else:
+            _cuda_path = cuda.get_rocm_path()
+
+    if not _is_hip:
+        if 9020 <= _cuda_runtime_version < 9030:
+            bundled_include = 'cuda-9.2'
+        elif 10000 <= _cuda_runtime_version < 10010:
+            bundled_include = 'cuda-10.0'
+        elif 10010 <= _cuda_runtime_version < 10020:
+            bundled_include = 'cuda-10.1'
+        elif 10020 <= _cuda_runtime_version < 10030:
+            bundled_include = 'cuda-10.2'
+        elif 11000 <= _cuda_runtime_version < 11010:
+            bundled_include = 'cuda-11.0'
+        elif 11010 <= _cuda_runtime_version < 11020:
+            bundled_include = 'cuda-11.1'
+        else:
+            # CUDA versions not yet supported.
+            bundled_include = None
 
-    if bundled_include is not None:
-        options += ('-I' + os.path.join(
-            _get_header_dir_path(), 'cupy', '_cuda', bundled_include),)
+        if bundled_include is None and _cuda_path is None:
+            raise RuntimeError(
+                'Failed to auto-detect CUDA root directory. '
```

Comment:
This message is CUDA specific and can lead to confusion to HIP users :)

Reply:
This branch is CUDA-only 😁 But let me fix the error message, as I messed up during merging...

Reply:
Done in 3e48758.